File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt user for text
    string text = get_string("Text: ");

    // Implementing all count functions
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Grade text inputted by user
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    // Calculate Coleman-Liau index
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    // Print grade
    int grade = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
        printf("Grade %d\n", grade);
}

int count_letters(string text)
{
    int letters = 0;
    // Count letters from user input
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    // Count words from user input
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    // Count sentences from user input by counting ! . ?
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
from cs50 import get_string

# Prompt user for text
text = get_string("Text: ")

# Count letters
letters = len([let for let in text if let.isalpha()])

# Count Words
words = len(text.split())

# Count sentences
full_stop = text.count(".")
excalmation = text.count("!")
question = text.count("?")
sentences = full_stop + excalmation + question

# Grade text
L = letters / words * 100
S = sentences / words * 100

# Calculate Coleman-Liau index
grade = round(0.0588 * L - 0.296 * S - 15.8)

# Print Grade
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
